File: javascripts/clean.js
import { InitLogger, Logger } from './src/logger.js';
import { Connect, CloseDB, ExecuteSQLWithParams } from './src/database.js';

const main = async () => {
    try {
        InitLogger();
        await Connect();

        Logger.info("Starting database cleanup...");

        await ExecuteSQLWithParams("DROP TABLE IF EXISTS DemoAccount");

        Logger.info("Successfully dropped table DemoAccount.");
        Logger.info("Cleanup complete.");

        await CloseDB();
    } catch (err) {
        Logger.error("Error during cleanup:", { error: err.message });
    }
};

await main();

File: golang/cmd/clean/clean.go
package main

import (
	"log/slog"

	"yogabyte-db-connection/src/database"
	"yogabyte-db-connection/src/logger"
)

func main() {
	// Initialize Logger
	logger.InitLogger()

	// Initialize Database
	if err := database.Connect(); err != nil {
		slog.Error("Failed to initialize database", "error", err)
		panic(err)
	}
	defer database.CloseDB()

	slog.Info("Starting database cleanup...")

	// Execute Drop
	rows, err := database.ExecuteSQLWithParams(`DROP TABLE IF EXISTS DemoAccount`)
	if err != nil {
		slog.Error("Failed to drop table", "error", err)
		panic(err)
	}
	rows.Close() // Ensure connection is released

	slog.Info("Successfully dropped table DemoAccount.")
	slog.Info("Cleanup complete.")
}
